### ISSUE TITLE
Installation problems

### ISSUE THREAD

**Issue by mleipold** [JinmiaoChenLab/cytofkit#1]
I'm trying to install the cytofkit software.  I am running Mac OSX 10.10.3, XQuartz 2.7.9 (freshly updated during installation attempts), and R 3.2.3 (or RStudio Version 0.98.1103).
## I was able to download it from Bioconductor, and it looked like it installed:

> source("https://bioconductor.org/biocLite.R")
> Bioconductor version 3.1 (BiocInstaller 1.18.5), ?biocLite for help
> A newer version of Bioconductor is available for this version of R, ?BiocUpgrade for help
> biocLite("cytofkit")
> BioC_mirror: http://bioconductor.org
> Using Bioconductor version 3.1 (BiocInstaller 1.18.5), R version 3.2.3.
> Installing package(s) ‘cytofkit’
> trying URL 'http://bioconductor.org/packages/3.1/bioc/bin/macosx/mavericks/contrib/3.2/cytofkit_1.1.0.tgz'
> # Content type 'application/x-gzip' length 1812061 bytes (1.7 MB)
> 
> downloaded 1.7 MB

The downloaded binary packages are in
    /var/folders/ln/_r_qkh8j1wz_wlrx0gw3mp100000gq/T//RtmpAu0vRJ/downloaded_packages
Old packages: 'manipulate'
Update all/some/none? [a/s/n]: 
a
trying URL 'http://cran.rstudio.com/bin/macosx/mavericks/contrib/3.2/manipulate_1.0.1.tgz'
# Content type 'application/x-gzip' length 34143 bytes (33 KB)

downloaded 33 KB

The downloaded binary packages are in
##     /var/folders/ln/_r_qkh8j1wz_wlrx0gw3mp100000gq/T//RtmpAu0vRJ/downloaded_packages

I can see "cytofkit" as a package in RStudio.  As shown above, a few packages had to be updated, but all that happened fine as far as I can tell.  Similarly, during the first installation attempt, XQuartz wanted an update, so I did that.  I even restarted my computer after the XQuartz update to make sure that wouldn't be an issue.

When I move on to the next step:

> library("cytofkit")

It pauses, XQuartz opens, and then I get the ">" back.

After that, though, I can't get anything to work.  

---

> ?cytofkit-package
> Error in eval(expr, envir, enclos) : object 'cytofkit' not found
> Error in .helpForCall(topicExpr, parent.frame()) : 
>   error in trying to evaluate the expression for argument ‘e1’ (cytofkit)
> cytofkit_GUI()
> Error: could not find function "cytofkit_GUI"
> ---

In short:  the GUI never opens...it seems like there's an error earlier in the process.

Suggestions?

-Mike, HIMC

**Comment by ghowell2812** [comment]
Same here - have managed to download everything fine but when i try and launch the GUI with cytofkit_GUI() i get the following messahe and nothing progresses

library("cytofkit")
> cytofkit_GUI()
Error in structure(.External(.C_dotTclObjv, objv), class = "tclObj") : 
  [tcl] invalid command name "toplevel".

And if i try to query the package help i get 

> ?`cytofkit-package`
Error in eval(expr, envir, enclos) : object 'cytofkit-package' not found
Error in .helpForCall(topicExpr, parent.frame()) : 
  error in trying to evaluate the expression for argument ‘e2’ (cytofkit-package)

I've tried reinstalling and i think i have all of the dependancies inplace

Thanks, Gareth

**Comment by jinmiaochen** [comment]
Hi Gareth,

Try the following without the quote?
library(cytofkit)

Best,
Jinmiao

**Comment by jinmiaochen** [comment]
Hi Mike,

Maybe you would like to install the current release of Bioconductor version 3.3 ? It works with R version 3.3.0 and above. We keep upgrading cytofkit and have added several lovely new features to the latest release.

Good luck!

Best,
Jinmiao

**Comment by haoeric** [comment]
Dear @mleipold,

The latest version of [cytofkit](https://www.bioconductor.org/packages/release/bioc/html/cytofkit.html) (version 1.4.8) is on bioconductor version 3.3, which requires R version higher than 3.3.0. So could you update your [R](https://www.r-project.org) and then install cytofkit with codes below:

```R
source("https://bioconductor.org/biocLite.R")
biocLite("cytofkit")
```
Thanks for your patience, and please let us know if you have any other questions, thanks!

**Comment by haoeric** [comment]
Dear @ghowell2812,

Thanks for your feedback. From your information, I may guess that cytofkit is not installed properly on your side. Could you check your R version (type `version` on R console will print your R version) and make sure it is version 3.3.0 or above. Then update BiocInstaller and install cytofkit with codes below:

```R
source("https://bioconductor.org/biocLite.R")
biocLite("BiocInstaller")

source("https://bioconductor.org/biocLite.R")
biocLite("cytofkit")
```

**Comment by ghowell2812** [comment]
Dear haoeric,

I tried the following and still the same issue from what i can see

> version
               _                           
platform       x86_64-apple-darwin13.4.0   
arch           x86_64                      
os             darwin13.4.0                
system         x86_64, darwin13.4.0        
status                                     
major          3                           
minor          3.1                         
year           2016                        
month          06                          
day            21                          
svn rev        70800                       
language       R                           
version.string R version 3.3.1 (2016-06-21)
nickname       Bug in Your Hair            
> source("https://bioconductor.org/biocLite.R")
Bioconductor version 3.3 (BiocInstaller 1.22.3), ?biocLite for help
> biocLite("BiocInstaller")
BioC_mirror: https://bioconductor.org
Using Bioconductor 3.3 (BiocInstaller 1.22.3), R 3.3.1 (2016-06-21).
Installing package(s) ‘BiocInstaller’
trying URL 'https://bioconductor.org/packages/3.3/bioc/bin/macosx/mavericks/contrib/3.3/BiocInstaller_1.22.3.tgz'
Content type 'application/x-gzip' length 54312 bytes (53 KB)
==================================================
downloaded 53 KB


The downloaded binary packages are in
	/var/folders/3b/ltp_tk5j193688__dyy4k2yw0000gq/T//RtmpuUqRkw/downloaded_packages
Old packages: 'R6'
Update all/some/none? [a/s/n]: 
a

  There is a binary version available but the source version is later:
   binary source needs_compilation
R6  2.1.3  2.2.0             FALSE

installing the source package ‘R6’

trying URL 'https://cran.rstudio.com/src/contrib/R6_2.2.0.tar.gz'
Content type 'application/x-gzip' length 203498 bytes (198 KB)
==================================================
downloaded 198 KB

* installing *source* package ‘R6’ ...
** package ‘R6’ successfully unpacked and MD5 sums checked
** R
** inst
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (R6)

The downloaded source packages are in
	‘/private/var/folders/3b/ltp_tk5j193688__dyy4k2yw0000gq/T/RtmpuUqRkw/downloaded_packages’
> source("https://bioconductor.org/biocLite.R")
Bioconductor version 3.3 (BiocInstaller 1.22.3), ?biocLite for help
> biocLite("cytofkit")
BioC_mirror: https://bioconductor.org
Using Bioconductor 3.3 (BiocInstaller 1.22.3), R 3.3.1 (2016-06-21).
Installing package(s) ‘cytofkit’
trying URL 'https://bioconductor.org/packages/3.3/bioc/bin/macosx/mavericks/contrib/3.3/cytofkit_1.4.8.tgz'
Content type 'application/x-gzip' length 8639750 bytes (8.2 MB)
==================================================
downloaded 8.2 MB


The downloaded binary packages are in
	/var/folders/3b/ltp_tk5j193688__dyy4k2yw0000gq/T//RtmpuUqRkw/downloaded_packages
> library(cytofkit)
Loading required package: ggplot2
Loading required package: plyr
Warning message:
In fun(libname, pkgname) : couldn't connect to display ":0"

> cytofkit_GUI()
Error in structure(.External(.C_dotTclObjv, objv), class = "tclObj") : 
  [tcl] invalid command name "toplevel".

I downloaded the R6 update as suggested in the log and still the same issue when i use cytofkit_GUI(). And it happens with library("cytofkit") and library(cytofkit)

Gareth

**Comment by mleipold** [comment]
OK.  I updated to R version 3.3.1 (2016-06-21) -- "Bug in Your Hair"

I then re-downloaded "cytofkit" as described.  After a LOT of dependency downloads and updates, it finally finished.
----
> biocLite("cytofkit")
BioC_mirror: https://bioconductor.org
Using Bioconductor 3.3 (BiocInstaller 1.22.3), R 3.3.1 (2016-06-21).
Installing package(s) ‘cytofkit’
also installing the dependencies ‘minqa’, ‘nloptr’, ‘lme4’, ‘SparseM’, ‘MatrixModels’, ‘zoo’, ‘DEoptimR’, ‘chron’, ‘pbkrtest’, ‘quantreg’, ‘lmtest’, ‘RColorBrewer’, ‘dichromat’, ‘munsell’, ‘labeling’, ‘robustbase’, ‘mvtnorm’, ‘pcaPP’, ‘bitops’, ‘stringi’, ‘Formula’, ‘latticeExtra’, ‘acepack’, ‘gridExtra’, ‘data.table’, ‘colorspace’, ‘car’, ‘sp’, ‘vcd’, ‘laeken’, ‘ALL’, ‘RUnit’, ‘pkgmaker’, ‘registry’, ‘rngtools’, ‘gridBase’, ‘digest’, ‘gtable’, ‘scales’, ‘BiocGenerics’, ‘graph’, ‘rrcov’, ‘corpcor’, ‘matrixStats’, ‘BH’, ‘gtools’, ‘gdata’, ‘caTools’, ‘stringr’, ‘httpuv’, ‘mime’, ‘jsonlite’, ‘xtable’, ‘htmltools’, ‘R6’, ‘sourcetools’, ‘permute’, ‘foreach’, ‘iterators’, ‘RcppEigen’, ‘Hmisc’, ‘FNN’, ‘VIM’, ‘proxy’, ‘scatterplot3d’, ‘ConsensusClusterPlus’, ‘tsne’, ‘flowUtils’, ‘XML’, ‘magrittr’, ‘NMF’, ‘irlba’, ‘ggplot2’, ‘plyr’, ‘Rtsne’, ‘e1071’, ‘flowCore’, ‘gplots’, ‘VGAM’, ‘reshape2’, ‘ggrepel’, ‘shiny’, ‘vegan’, ‘Biobase’, ‘doParallel’, ‘pdist’, ‘destiny’, ‘FlowSOM’, ‘igraph’, ‘RANN’, ‘Rcpp’
-----
There were a couple warnings along the way:
-----
clang++ -std=c++11 -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -I. -Iicu55 -Iicu55/unicode -Iicu55/common -Iicu55/i18n -DU_STATIC_IMPLEMENTATION -DU_COMMON_IMPLEMENTATION -DU_I18N_IMPLEMENTATION -DUCONFIG_USE_LOCAL  -UDEBUG -DNDEBUG -I/usr/local/include -I/usr/local/include/freetype2 -I/opt/X11/include   -fPIC -fPIC  -Wall -mtune=core2 -g -O2 -c icu55/common/unistr.cpp -o icu55/common/unistr.o
icu55/common/unistr.cpp:1798:13: warning: unused function 'uprv_UnicodeStringDummy' [-Wunused-function]
static void uprv_UnicodeStringDummy(void) {
            ^
1 warning generated.
clang++ -std=c++11 -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -I. -Iicu55 -Iicu55/unicode -Iicu55/common -Iicu55/i18n -DU_STATIC_IMPLEMENTATION -DU_COMMON_IMPLEMENTATION -DU_I18N_IMPLEMENTATION -DUCONFIG_USE_LOCAL  -UDEBUG -DNDEBUG -I/usr/local/include -I/usr/local/include/freetype2 -I/opt/X11/include   -fPIC -fPIC  -Wall -mtune=core2 -g -O2 -c icu55/common/unistr_props.cpp -o icu55/common/unistr_props.o
-----
After that was done, I loaded cytofkit:
> library("cytofkit")
Loading required package: ggplot2
Loading required package: plyr
> ?cytofkit-package
Error in eval(expr, envir, enclos) : object 'package' not found
Error in .helpForCall(topicExpr, parent.frame()) : 
  error in trying to evaluate the expression for argument ‘e2’ (package)
> cytofkit_GUI()  

In short:  the GUI now works.  I'll let you know if I have further issues.

However, your "?cytofkit-package" command is still failing.  Are you sure that's correct?


Mike

**Comment by haoeric** [comment]
Hi @ghowell2812, I think now cytofkit package is installed properly on your mac. The problem happens when calling the GUI, could you check if [xquartz](https://www.xquartz.org) is installed properly (pity that mac need this tool to support tcltk). 

You can try codes below:

```
library(tcltk)
```
If there is an error, please install [xquartz](https://www.xquartz.org) and then try again.

Thanks for your patience.

**Comment by haoeric** [comment]
COOL, @mleipold 

Sorry for my mistake, it should be:

```
?"cytofkit-package"
```

**Comment by ghowell2812** [comment]
Hi @haoeric turns out I have to open XQuartz first before opening R. Which is fine, seems to be fine now. Thanks for your help. G

**Comment by mleipold** [comment]
hi @haoeric 

thanks for the update; with the addition of the quotes, I now get a popup window of the RHelp documentation.  I would suggest that you correct the documentation (https://www.bioconductor.org/packages/release/bioc/vignettes/cytofkit/inst/doc/cytofkit_example.html)

@ghowell2812 that's odd that you have to open XQuartz first.  when I do it, the library command automatically opens XQuartz:
----
> library("cytofkit")
Loading required package: ggplot2
Loading required package: plyr
---- 
It sits there after "plyr" and flashes for a few seconds while XQuartz opens, but then I get my ">" back.  What version of R, and what version of XQuartz do you have?

**Comment by haoeric** [comment]
@ghowell2812, happy to know it works for you now, we welcome suggestions and feedbacks.

**Comment by haoeric** [comment]
Hi @mleipold Yes, we will correct the vignette in next version, thanks.

**Comment by ghowell2812** [comment]
@haoeric @mleipold when i run it on my iMac (R v3.3.1, XQuartz v2.7.9, ElCapitan v10.11.2) it works fine and starts up as it should. On my macbook (same version of R and XQuartz, ElCapitan v10.11.5) i have to launch XQuartz first...go figure!! Don't want to upgrade to Sierra just yet..!
At least it works now on both and looks really good. G

**Comment by MattMyint** [comment]
Seeing as this is almost a year old, I'll be closing this issue. Feel free to reopen if there's any other problems
